packages/@uppy/companion/src/server/emitter/redis-emitter.js

```diff
@@ -1,15 +1,46 @@
-const NRP = require('node-redis-pubsub')
+const redis = require('redis')
 
 /**
  * This module simulates the builtin events.EventEmitter but with the use of redis.
  * This is useful for when companion is running on multiple instances and events need
  * to be distributed across.
  */
 module.exports = (redisUrl, redisPubSubScope) => {
-  const nrp = new NRP({ url: redisUrl, scope: redisPubSubScope })
+  const prefix = redisPubSubScope ? `${redisPubSubScope}:` : ''
+  const publisher = redis.createClient({ url: redisUrl })
+  let subscriber
+
+  const connectedPromise = publisher.connect().then(() => {
+    subscriber = publisher.duplicate()
+    return subscriber.connect()
+  })
+  connectedPromise.catch(() => {}) // to avoid crashing in case of rejection.
```

Comment:
what will actually happen if we get an error here? I assume it will be some kind of connection error. Will companion continue to run as if there was no error?

Reply:
At line 17, we cannot get an error (`.catch(() => {})` will swallow any error that could came up). `connectedPromise` will stay in its rejected state though, so any further call will either call the `errorHandler` if it's been set, or send a `unhandledRejection` error (that will crash the process on Node.js 16+)